### PR TITLE
Updates/dataset factory updater

### PR DESCRIPTION
**Pull request by dpopleton**
This update fixes several small issues:
species taxonomy id will be populated and common name will be present.
It also uses the dataset factory for dataset creation and creates homology datasets as well.
The previously broken tests have been fixed, but will need future work as they are interdependent.

### CODE REVIEW

File: src/tests/databases/core_8/meta.txt [hunk 1]
```diff
@@ -19,4 +19,5 @@
 19	1	genebuild.sample_location	KB871578.1:9766653-9817473
 20	1	strain.type	test
 23	1	genebuild.provider_name	test
-24	1	genebuild.start_date	test
\ No newline at end of file
+24	1	genebuild.start_date	test
```
Comment (marcoooo) [inline src/tests/databases/core_8/meta.txt:22]: Start_date in test.... should be a date. at least.

File: src/tests/databases/core_9/meta.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-12	1	assembly.accession	GCA_000002985.3
+12	1	assembly.accession	test1
```
Comment (marcoooo) [inline src/tests/databases/core_9/meta.txt:1]: Why changing these values?

File: src/tests/databases/core_9/meta.txt [hunk 3]
```diff
@@ -14,11 +14,12 @@
 2	1	species.taxonomy_id	6239
 10	1	species.type	monsters
 5	1	species.url	Jabbe
-17	1	genebuild.version	EXT01
+17	1	genebuild.version	ENS01
```
Comment (marcoooo) [inline src/tests/databases/core_9/meta.txt:17]: Don't get why this PR change so much test data.

File: src/tests/databases/core_9/meta.txt [hunk 4]
```diff
@@ -14,11 +14,12 @@
...
 18	1	genebuild.sample_gene	ENSAMXG00005000318
 19	1	genebuild.sample_location	KB871578.1:9766653-9817473
 20	1	strain.type	test
 23	1	genome.genome_uuid	a733550b-93e7-11ec-a39d-005056b38ce3
 24	1	genebuild.havana_datafreeze_date	test2
 25	1	assembly.total_genome_length	546
-26	1	genebuild.start_date	test
\ No newline at end of file
+26	1	genebuild.start_date	test
```
Comment (marcoooo) [inline src/tests/databases/core_9/meta.txt:24]: Start date should be a date

**Review by marcoooo — CHANGES_REQUESTED** [review]

File: src/ensembl/production/metadata/api/factories/datasets.py [hunk 5]
```diff
@@ -30,8 +30,18 @@ def create_all_child_datasets(self, session, dataset_uuid):
         top_level_dataset = self.__get_dataset(session, dataset_uuid)
         self.__create_child_datasets_recursive(session, top_level_dataset)
 
-    def create_dataset(self, session, genome_uuid, dataset_source, dataset_type, dataset_attributes, name, label,
-                       version):
+    def create_dataset(self, session, genome_input, dataset_source, dataset_type, dataset_attributes, name, label,
```
Comment (vinay-ebi) [inline src/ensembl/production/metadata/api/factories/datasets.py:33]: i it  genome_uuid ?

File: VERSION [hunk 6]
```diff
@@ -1 +1 @@
-2.0.1
```
[the remark this answers is not in the document]
Comment (dpopleton) [reply VERSION:1]: Yes

File: src/ensembl/production/metadata/updater/core.py [hunk 7]
```diff
@@ -91,7 +88,7 @@ def process_species(self, species_id):
         This method contains the logic for updating the metadata
         """
 
-        with self.metadata_db.session_scope() as meta_session:
+        with (self.metadata_db.session_scope() as meta_session):
```
[the remark this answers is not in the document]
Comment (dpopleton) [reply src/ensembl/production/metadata/updater/core.py:91]: Removed

File: src/ensembl/production/metadata/updater/core.py [hunk 8]
```diff
@@ -153,6 +150,35 @@ def process_species(self, species_id):
...
+                                                   ).join(attribute_alias2, dataset_attr_alias2.attribute
+                                                          ).filter(Assembly.accession == assembly.accession,
+                                                                   Dataset.dataset_type.has(name="genebuild"),
+                                                                   and_(
+                                                                       attribute_alias1.name == "genebuild.provider_name",
+                                                                       dataset_attr_alias1.value == provider_name,
+                                                                       attribute_alias2.name == "genebuild.last_geneset_update",
+                                                                       dataset_attr_alias2.value == geneset_update
+                                                                   )
+                                                                   )
```
[the remark this answers is not in the document]
Comment (dpopleton) [reply src/ensembl/production/metadata/updater/core.py:177]: We said that if a geneset is to be udpdated, there must be either a new provider_name or last_geneset_update.
I had trouble keeping this one small.

File: src/tests/databases/core_9/meta.txt (hunk 3 above)
Comment (dpopleton) [reply src/tests/databases/core_9/meta.txt:17]: We had 12 assert statements commented out. They pass now.

File: src/tests/databases/core_9/meta.txt (hunk 2 above)
Comment (dpopleton) [reply src/tests/databases/core_9/meta.txt:1]: This is the interdependent issue. On a previous update you swapped out the values for real ones and commented out the tests. These are changed as the tests were written to see if a previously created dataset is overridden. A rewrite is possible but not trivial for these to get proper values and independence incorporated.

File: src/tests/databases/core_8/meta.txt (hunk 1 above)
Comment (dpopleton) [reply src/tests/databases/core_8/meta.txt]: Replaced with existing data

File: src/tests/databases/core_9/meta.txt (hunk 4 above)
Comment (dpopleton) [reply src/tests/databases/core_9/meta.txt]: Replaced with 2023-07-Ensembl in all cases

File: src/tests/databases/core_9/meta.txt (hunk 2 above)
Comment (marcoooo) [reply src/tests/databases/core_9/meta.txt:1]: So the test that has been reactivated should either be inside the test on which it depends. or commented out again. These dependencies cause more issues, and are very error prone.

**Review by marcoooo — APPROVED** [review]

**Pull request merged**